estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,59 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+
+
+class PropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Real Estate offer"
+    _sql_constraints = [
+        ('check_price', 'CHECK(price > 0)',
+         'A property offer must be strictly positive.'),
+    ]
+
+    price = fields.Float()
+    status = fields.Selection(
+        copy=False, selection=[("accepted", "Accepted"), ("refused", "Refused")]
+    )
+    partner_id = fields.Many2one("res.partner", required=True, string="Buyer")
+    property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7, string="Validity (days)")
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline", inverse="_compute_validity", string="Deadline"
+    )       
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            created_at = record.create_date or fields.Date.today()
+            record.date_deadline = fields.Date.add(created_at, days=record.validity)
+
+    def _compute_validity(self):
+        for record in self:
+            created_at = record.create_date or fields.Date.today()
+            delta = record.date_deadline - created_at.date()
+            record.validity = delta.days
+
+    def action_confirm(self):
+        for record in self:
+            if record.property_id.state in ["sold", "canceled"]:
+                raise UserError("Property reached a final state")
+            if record.property_id.state in ["Offer_accepted"]:
```

Comment:
```suggestion
            if record.property_id.state == "offer_accepted":
```
I know it's the same, but there is no use to check in a list if there is only a single element. Also I think you put an uppercase while your selection choice has a lowercase